### PR TITLE
Set integration name for Homebrew installs

### PR DESCRIPTION
**Pull request by maxjeffos**
- [X] Ready for review
- [X] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Adds the integration name`HOMEBREW` when the running CLI instances is installed with Homebrew.

#### What are the relevant tickets?
HAMMER-135

### CODE REVIEW

File: src/lib/analytics-sources.ts [hunk 1]
```diff
@@ -114,3 +118,27 @@ export function validateScoopManifestFile(snykExecutablePath: string): boolean {
   }
   return false;
 }
+
+export function isHomebrew(): boolean {
+  const currentProcessPath = process.execPath;
+  const isHomebrewPath = currentProcessPath.includes('Cellar/snyk');
```
Comment (JackuB) [inline src/lib/analytics-sources.ts:124]: I'd consider extending the match

```suggestion
  const isHomebrewPath = currentProcessPath.includes('/Cellar/snyk/');
```

File: test/analytics-sources.spec.ts [hunk 2]
```diff
@@ -47,6 +51,62 @@ describe('analytics-sources - scoop detection', () => {
   });
 });
 
+describe('analytics-sources - Homebrew detection', () => {
+  afterEach(() => {
+    // jest.resetAllMocks();
```
Comment (JackuB) [inline test/analytics-sources.spec.ts:56]: Some leftovers?

File: test/analytics-sources.spec.ts [hunk 3]
```diff
@@ -47,6 +51,62 @@ describe('analytics-sources - scoop detection', () => {
...
+      expect(isHomebrew()).toBe(false);
+      expect(fileExistsSpy).not.toHaveBeenCalled();
+    } finally {
+      process.execPath = originalExecPath;
+    }
+  });
+
+  it('returns false if formula files does not exist', () => {
+    const originalExecPath = process.execPath;
+    try {
```
Comment (JackuB) [inline test/analytics-sources.spec.ts:88]: Is it worth try catching? If it throws, we would want the test to fail, no?
Same for the other tests that are using try/catch

**Review by JackuB — APPROVED** [review]
Small comment on the test format, but looks solid 👍

File: test/analytics-sources.spec.ts (hunk 2 above)
Comment (maxjeffos) [reply test/analytics-sources.spec.ts:56]: :green_tick:

File: src/lib/analytics-sources.ts (hunk 1 above)
Comment (maxjeffos) [reply src/lib/analytics-sources.ts:124]: good idea - done. thanks

File: test/analytics-sources.spec.ts (hunk 3 above)
Comment (maxjeffos) [reply test/analytics-sources.spec.ts:88]: Ya you're right. I thought it was nice to have the reset stuff separated into a finally like this to make it more obvious but effectively it's the same and just takes up more space.

**Pull request merged**